R/data-objects.R

```diff
@@ -56,24 +62,113 @@ iput <- function(
   stop_irods_overwrite(overwrite, logical_path)
 
   # what type of object are we dealing with
-  if (is.character(x) && file.exists(x)) {
-    fil <- x
-  } else if (is.symbol(substitute(x))) {
+  if (!(is.character(local_path) && file.exists(local_path))) {
+    stop("Local file does not exist.", call. = FALSE)
+  }
+
```

Comment:
I would turn this check into two if statements (I can take care of that in a future commit, though, it's not an urgent thing).

```r
if (!is.character(local_path) || length(local_path) > 1) {
    stop("Please provide a character vector of length 1 as local path", call. = FALSE)
} else if (!file.exists(local_path)) {
   stop(sprintf("Local file '%s' does not exist.", local_path), call. = FALSE)
}
```

Reply:
I have not touched this now. Ps, you can also use `stop()` as `paste0()` would function.
```
   stop("Local file ", local_path, " does not exist.", call. = FALSE)
```